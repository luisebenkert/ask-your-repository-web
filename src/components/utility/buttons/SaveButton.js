// @flow
import React from 'react';
import type { Node } from 'react';
import classNames from 'classnames';
import Button from './Button';

type Props = {
  children: Node,
  className?: string,
};

function SaveButton({
  children,
  className,
  ...rest
}: Props) {
  const styleClasses = classNames('Button--save', className);

  return (
    <Button
      className={styleClasses}
      data-cy="save-button"
      {...rest}
    >
      {children}
    </Button>
  );
}

SaveButton.defaultProps = {
  className: '',
};

export default SaveButton;
