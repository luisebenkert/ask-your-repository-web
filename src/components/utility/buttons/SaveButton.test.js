// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SaveButton from './SaveButton';

describe('<SaveButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow((
      <SaveButton>
        Save
      </SaveButton>
    ));
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
