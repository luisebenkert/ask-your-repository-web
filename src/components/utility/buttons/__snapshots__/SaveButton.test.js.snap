// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveButton /> renders correctly 1`] = `
<Button
  className="Button--save"
  data-cy="save-button"
  type="button"
>
  Save
</Button>
`;
